ballerina/channel_csv_io.bal

```diff
@@ -135,3 +161,43 @@ isolated function getWritableCSVChannel(WritableChannel writableChannel) returns
     }
     return writableCSVChannel;
 }
+
+isolated function readHeadersFromCsvFile(string path) returns string[]|Error {
+    stream<string[], Error?>|Error csvContent = fileReadCsvAsStream(path);
+        if csvContent is Error {
+            if csvContent is FileNotFoundError {
+                return [];
+            } else {
+                return error GenericError((<Error> csvContent).message());
+            }
+        } else {
+            do {
+                var csvLine = csvContent.next();
+                if (csvLine is Error) {
+                    check error GenericError((<Error> csvLine).message());
+                } else if csvLine !is () {
+                    if csvLine["value"] != [""] {
+                        return csvLine["value"];
+                    }
```

Comment:
Do we need to do this here?. doesn't this handle in header validation?

Reply:
If the input is an empty csv file (a file even without any headers) [""] is the input value we get. so this has to be checked here

Reply:
If it is an empty csv file, `header[] == [""]`. Here we are not validating the header right? we just pass what we get and do the validation in the other function.